### PR TITLE
[BFCL] Add New Model `o1-preview-2024-09-12` and `o1-mini-2024-09-12` 

### PR DESCRIPTION
**Pull request by HuanzhiMao**
This PR adds new models `o1-preview-2024-09-12` and `o1-mini-2024-09-12` to the leaderboard.

### CODE REVIEW

File: berkeley-function-call-leaderboard/bfcl/model_handler/proprietary_model/openai.py [hunk 1]
```diff
@@ -31,16 +33,33 @@ def inference(self, prompt, functions, test_category):
             prompt = system_prompt_pre_processing_chat_model(
                 prompt, DEFAULT_SYSTEM_PROMPT, functions
             )
+            # Special handling for o1-preview and o1-mini as they don't support system prompts yet
+            if "o1-preview" in self.model_name or "o1-mini" in self.model_name:
+                prompt = convert_system_prompt_into_user_prompt(prompt)
+                prompt = combine_consecutive_user_prompr(prompt)
```
Comment (CharlieJCJ) [inline berkeley-function-call-leaderboard/bfcl/model_handler/proprietary_model/openai.py:39]: s/prompr/prompt

File: berkeley-function-call-leaderboard/bfcl/model_handler/proprietary_model/openai.py [hunk 2]
```diff
@@ -31,16 +33,33 @@ def inference(self, prompt, functions, test_category):
...
-            )
+            # These two models have temperature and top_p fixed to 1, and max_tokens is not supported
+            # Beta limitation: https://platform.openai.com/docs/guides/reasoning/beta-limitations
+            if "o1-preview" in self.model_name or "o1-mini" in self.model_name:
+                # Rate limit workaround, 20 requests per minute even for top tier users
+                time.sleep(4)
+                response = self.client.chat.completions.create(
+                    messages=message,
+                    model=self.model_name,
+                    temperature=1,
```
Comment (CharlieJCJ) [inline berkeley-function-call-leaderboard/bfcl/model_handler/proprietary_model/openai.py:49]: Why temperature=1?

Comment (HuanzhiMao) [reply berkeley-function-call-leaderboard/bfcl/model_handler/proprietary_model/openai.py:49]: https://github.com/ShishirPatil/gorilla/pull/635/files#diff-00e05c9b42aba3f5a159dd93255f9be26b0061463815b62469e6a2be2be6d108R44

**Review by CharlieJCJ — APPROVED** [review]
LGTM

**Pull request merged**